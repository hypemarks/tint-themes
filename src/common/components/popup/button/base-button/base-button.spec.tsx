import { render, screen } from '@testing-library/react';
import React from 'react';
import BaseButton from './base-button';
import userEvent from '@testing-library/user-event';

describe('Test <BaseButton /> Component', () => {
  it('should render snapshot', () => {
    const wrapper = render(<BaseButton onClick={jest.fn} />);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });

  it('should called onClick callback', () => {
    const onclick = jest.fn();
    render(<BaseButton onClick={onclick} />);
    userEvent.click(screen.getByTestId('base-button'));
    expect(onclick).toHaveBeenCalledTimes(1);
  });

  it('should render with correct class and styles', () => {
    const className = 'className';
    const styles = { left: '20' };
    const wrapper = render(
      <BaseButton onClick={jest.fn} primary={true} extended={true} className={className} customStyles={styles} />
    );
    const button = wrapper.queryByTestId('base-button');
    expect(button).toHaveClass('base-button--primary', 'base-button--extended', className);
    expect(button).toHaveStyle(styles);
  });
});
