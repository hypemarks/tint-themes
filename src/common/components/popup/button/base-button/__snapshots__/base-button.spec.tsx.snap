// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <BaseButton /> Component should render snapshot 1`] = `
<DocumentFragment>
  <button
    class="base-button undefined "
    data-testid="base-button"
    type="button"
  />
</DocumentFragment>
`;
